src/Maestro/SubscriptionActorService/LocalGit.cs

```diff
@@ -42,17 +44,26 @@ public LocalGit(TemporaryFiles tempFiles, IConfiguration configuration, Operatio
         public async Task<string> GetPathToLocalGitAsync()
         {
             // Determine whether we need to do any downloading at all.
-            if (!string.IsNullOrEmpty(_gitExecutable))
+            if (!string.IsNullOrEmpty(_gitExecutable) && File.Exists(_gitExecutable))
             {
-                return _gitExecutable;
+                // We should also mke sure that the git executable that exists runs properly
+                try
+                {
+                    LocalHelpers.ExecuteGitCommand(_gitExecutable, "--version", _logger, Environment.CurrentDirectory);
+                    return _gitExecutable;
+                }
+                catch (DarcException)
+                { 
+                    _logger.LogWarning($"Something went wrong with the git executable at {_gitExecutable}. Downloading new version.");
```

Comment:
Should this also remove the directory if it exists?

Reply:
I was considering that... I'm worried about ruining some other state, but I think, yes. It probably should. Perhaps below, when we download, we can delete the directory before creating it (ie, here: https://github.com/dotnet/arcade-services/pull/1512/files#diff-235ac10b12ba77f37591038be404a75eec1b5495d2251b05fc492f3b035b19b8R80)

Reply:
Well that link certainly did not work

Reply:
Yeah, I know where you mean. I think that's a good place.